src/jsonrpc/api.rs

```diff
@@ -87,6 +87,41 @@ fn broadcast_spend(control: &DaemonControl, params: Params) -> Result<serde_json
     Ok(serde_json::json!({}))
 }
 
+fn list_coins(control: &DaemonControl, params: Option<Params>) -> Result<serde_json::Value, Error> {
+    let statuses_arg = params
+        .clone()
+        .and_then(|p| p.get(0, "statuses").cloned())
+        .and_then(|statuses| statuses.as_array().cloned());
+    let mut statuses: Vec<CoinStatus> = vec![];
+    if let Some(statuses_arg) = statuses_arg {
+        for status_arg in statuses_arg.iter() {
+            statuses.push(
+                status_arg
+                    .as_str()
+                    .and_then(CoinStatus::from_arg)
+                    .ok_or_else(|| Error::invalid_params("Invalid 'statuses' parameter."))?,
+            );
+        }
+    }
+    let outpoints = params
+        .and_then(|p| p.get(1, "outpoints").cloned())
+        .and_then(|op| op.as_array().cloned())
+        .and_then(|arr| {
+            arr.iter()
+                .map(|entry| {
+                    entry
+                        .as_str()
+                        .and_then(|e| bitcoin::OutPoint::from_str(e).ok())
+                })
+                .collect::<Option<Vec<bitcoin::OutPoint>>>()
```

Comment:
We shouldn't be silently ignoring these errors.

Reply:
Ah yes, I've updated this now.